unison-share-api/src/Unison/Server/Backend.hs

```diff
@@ -346,6 +374,20 @@ termListEntry codebase b0 r n = do
 
   pure $ TermEntry r n ot tag
 
+checkIsTestForBranch :: Branch0 m -> Referent -> Bool
+checkIsTestForBranch b0 r =
+  Metadata.hasMetadataWithType' r Decls.isTestRef $ Branch.deepTermMetadata b0
+
+checkIsTestForV2Branch :: Monad m => V2Branch.Branch m -> V2.Referent -> m Bool
+checkIsTestForV2Branch b r = do
+  -- TODO: Should V2Branch use some sort of relation here?
```

Comment:
The v2 branch structure is just nested maps, so this sort of lookup is slightly slower, I don't think it's a big deal in this case, but we could likely optimize this in the future.

Reply:
I think this function is kinda weird (at no fault of this PR).

In the V2 branch, metadata is on the (name, ref) pair, not just the ref. ...yet the "is test" metadata would either be on all namings of a ref, or none (right...?).

Maybe we could short circuit in the implementation here to make it more clear that once we find some name for the given ref, we can look at that metadata blob and need not look further.

Reply:
I agree this is pretty hacky, I'll replace this with a check on the type of the expression seeing if it matches `[Test.Result]` in the next PR 👍🏼